src/main/scala/com/gu/scanamo/DynamoFormat.scala

```diff
@@ -176,6 +177,17 @@ object DynamoFormat extends DerivedDynamoFormat {
     */
   implicit val shortFormat = xmap(coerceNumber(_.toShort))(_.toString)(numFormat)
 
+  // Thrift and therefore Scanamo-Scrooge provides a byte and binary types backed by byte and byte[]
+  // Since AttributeValue includes a ByteBuffer instance, creating byte formats backed by ByteBuffer
+
+  private val javaByteBufferFormat = attribute[java.nio.ByteBuffer](_.getB, "B")(_.withB)
+
+  private def coerceByteBuffer[B](f: ByteBuffer => B): ByteBuffer => Xor[DynamoReadError, B] =
+    coerce[ByteBuffer,B, IllegalArgumentException](f)
+
+  implicit val byteArrayFormat = xmap(coerceByteBuffer(_.array()))(a => ByteBuffer.wrap(a))(javaByteBufferFormat)
+
+  implicit val byteFormat = xmap(coerceByteBuffer(_.get(0)))(b => ByteBuffer.wrap(Array(b)))(javaByteBufferFormat)
```

Comment:
I'd be tempted to store individual bytes as a number, in the same way we do with Short, rather than in a ByteBuffer of one byte. Do you have a particular preference either way?